src/contracts/urm/URMCompoundTimelock.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.30;
+
+// Internal Libraries
+import {URMCore} from "contracts/URMCore.sol";
+import {ICompoundTimelockTarget} from "interfaces/ICompoundTimelockTarget.sol";
+
+/// @title URMCompoundTimelock
+/// @author [ScopeLift](https://scopelift.co)
+/// @notice Integrates URMCore with a Compound-style Timelock contract as the execution target.
+/// @dev This contract implements the rollback proposal lifecycle using Compound's Timelock interface.
+///      It interacts with the target using individual `queueTransaction`, `cancelTransaction`, and `executeTransaction`
+/// calls for each action.
+///      Usecase:
+///        - Use this contract when your system uses Compound-style Timelocks, often found in simpler or more minimal
+/// governance systems (e.g., Compound, Fuse, etc.).
+///      Key Differences from URMTimelockController:
+///       - Queues, cancels, and executes each transaction individually (per `queueTransaction`).
+///       - Computes rollback IDs using `keccak256(abi.encode(...))` of the batch parameters.
+///       - Uses the `delay()` function on the target contract to determine delay.
+///       - No batch operation support or salt-based identifiers — relies purely on encoded parameters and ETA.
+///      Requirements:
+///        - The `TARGET` must conform to the `ICompoundTimelockTarget` interface, compatible with Compound’s Timelock.
+/// @dev Source:
+/// https://github.com/OpenZeppelin/openzeppelin-contracts/blob/ba35d580f47ba90494eb9f3d26f58f7949b10c67/contracts/vendor/compound/ICompoundTimelock.sol
+contract URMCompoundTimelock is URMCore {
```

Comment:
oh, one more note. let's consider calling these contracts something else that doesn't end in Timelock or TimelockController. The reason is that these contracts are not Timelocks, they are URMs. when i see it get imported downstream, i get a little confused 🤷
i'd even go further and wish that "RollbackManager" was explicit in the name, but now i'm just navel gazing

Reply:
https://github.com/ScopeLift/governance-rollback/issues/47#issuecomment-3055292702